sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/project/load/LoadProjects.groovy

```diff
@@ -41,6 +50,32 @@ class LoadProjects implements LoadProjectsInput{
         }
     }
 
+    @Override
+    void loadUserProjectsWithSubscriptionsFor(Subscriber subscriber) {
+        try {
+            List<Project> projects = loadUserProjects()
+            loadSubscriptionInformationInto(projects, subscriber)
+            output.loadedProjects(projects)
+        } catch (DataSourceException dataSourceException) {
+            output.failedExecution(dataSourceException.getMessage())
+        } catch (Exception e) {
+            output.failedExecution("Could not load projects")
+            log.error("unexpected exception during project loading:", e)
+        }
+    }
 
+    private List<Project> loadUserProjects() {
+        List<Project> projects = dataSource.fetchUserProjects()
+        return projects
+    }
 
+    /**
+     * Loads subscription information into a list of projects
+     * @param projects
+     * @param subscriber
+     */
+    private void loadSubscriptionInformationInto(Iterable<Project> projects, Subscriber subscriber) {
```

Comment:
can you explain why this is important for this use case?

Reply:
because we want to have the information on whether a user is subscribed to a project in the project :)

Reply:
Does this answer your question?